pkg/probe/probes.go

```diff
@@ -96,54 +118,144 @@ func checkNodeReadiness(client client.Client, timeout time.Duration) error {
-	pingOutput, err := ping(defaultGw, defaultGwProbeTimeout)
+	return nil
+}
+func lookupRootNS(nameServer string, timeout time.Duration) error {
+	rootNS := "root-server.net"
+	r := &net.Resolver{
+		PreferGo: true,
+		Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+			d := net.Dialer{
+				Timeout: timeout,
+			}
+			return d.DialContext(ctx, network, net.JoinHostPort(nameServer, "53"))
+		},
+	}
+	// We use a closure to create a scope for defer here
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	_, err := r.LookupNS(ctx, rootNS)
 	if err != nil {
-		return errors.Wrapf(err, "error pinging external address after network reconfiguration -> output: %s, currentState: %s", pingOutput, currentState)
+		return errors.Wrapf(err, "failed looking up NS %s using name sever %s", rootNS, nameServer)
 	}
+	return nil
+}
 
-	err = checkApiServerConnectivity(apiServerProbeTimeout)
+func runDNS(client client.Client, timeout time.Duration) error {
+	currentStateAsGJson, err := currentStateAsGJson()
 	if err != nil {
-		return errors.Wrapf(err, "error checking api server connectivity after network reconfiguration -> currentState: %s", currentState)
+		return errors.Wrap(err, "failed retrieving current state to get name resolving config")
 	}
 
-	err = checkNodeReadiness(client, nodeReadinessProbeTimeout)
+	// Get the name servers at node since the ones at container are not accurate
+	runningServersGJsonPath := "dns-resolver.running.server"
+	runningNameServers := currentStateAsGJson.Get(runningServersGJsonPath).Array()
+	if len(runningNameServers) == 0 {
+		return fmt.Errorf("missing name servers at '%s' on %s", runningServersGJsonPath, currentStateAsGJson.String())
+	}
+
+	// Check that at least one of the name servers can resolve root server NS
+	resolved := false
+	for _, runningNameServer := range runningNameServers {
+		err = lookupRootNS(runningNameServer.String(), defaultDnsProbeTimeout)
+		if err != nil {
+			log.Info("WARNING cannot resolve root server NS", "err", err)
+		} else {
+			resolved = true
+			break
+		}
+	}
+	if !resolved {
+		return errors.Wrap(err, "failed checking DNS connectivity")
+	}
+	return nil
+}
+
+// Select will return the external connectivity probes that are working (ping and dns) and
+// the internal connectivity probes
+func Select(cli client.Client) []Probe {
+	probes := []Probe{}
+
+	err := runPing(cli, 1*time.Second)
+	if err == nil {
+		probes = append(probes, Probe{
+			name:    "ping",
+			timeout: defaultGwProbeTimeout,
+			run:     runPing,
+		})
+	} else {
+		log.Info("WARNING not selecting 'ping' probe")
+	}
+	err = runDNS(cli, 1*time.Second)
+	if err == nil {
+		probes = append(probes, Probe{
+			name:    "dns",
+			timeout: defaultDnsProbeTimeout,
+			run:     runDNS,
+		})
+	} else {
+		log.Info("WARNING not selecting 'dns' probe")
+	}
+
+	// This probes use it's own client to bypass cache that
```

Comment:
I don't think this comment is useful.

Plus there is a typo "its"

Reply:
Not the right place I think I moved it from the probe.